nilearn/tests/test_helpers.py

```diff
@@ -0,0 +1,127 @@
+    mock_input, replacement_params = _mock_args_for_testing_replace_parameter()
+    expected_output = {
+        'unchanged_param_0': 'unchanged_param_0_val',
+        'replacement_param_0': 'deprecated_param_0_val',
+        'replacement_param_1': 'deprecated_param_1_val',
+        'unchanged_param_1': 'unchanged_param_1_val',
+        }
+    actual_ouput = helpers._transfer_deprecated_param_vals(
+            replacement_params,
+            mock_input,
+            )
+    assert actual_ouput == expected_output
+    
+
+def test_future_warn_deprecated_params():
+    mock_input, replacement_params = _mock_args_for_testing_replace_parameter()
+    expected_warnings = [
+        ('The parameter "deprecated_param_0" will be removed in a future Nilearn version. Please use the parameter "replacement_param_0" instead.'),
+        ('The parameter "deprecated_param_1" will be removed in a future Nilearn version. Please use the parameter "replacement_param_1" instead.'),
+        ]
+    with warnings.catch_warnings(record=True) as raised_warnings:
+        helpers._warn_deprecated_params(
+                replacement_params,
+                end_version='future',
+                lib_name='Nilearn',
+                kwargs=mock_input,
+                )
+    for raised_warning_, expected_warning_ in zip(raised_warnings, expected_warnings):
+        assert str(raised_warning_.message) == expected_warning_
+
+
+def test_next_warn_deprecated_params():
+    mock_input, replacement_params = _mock_args_for_testing_replace_parameter()
+    expected_warnings = [
+        ('The parameter "deprecated_param_0" will be removed in the next Nilearn version. Please use the parameter "replacement_param_0" instead.'),
+        ('The parameter "deprecated_param_1" will be removed in the next Nilearn version. Please use the parameter "replacement_param_1" instead.'),
+        ]
+    with warnings.catch_warnings(record=True) as raised_warnings:
+        helpers._warn_deprecated_params(
+                replacement_params,
+                end_version='next',
+                lib_name='Nilearn',
+                kwargs=mock_input,
+                )
+    for raised_warning_, expected_warning_ in zip(raised_warnings, expected_warnings):
+        assert str(raised_warning_.message) == expected_warning_
+
+
+def test_version_warn_deprecated_params():
+    mock_input, replacement_params = _mock_args_for_testing_replace_parameter()
+    expected_warnings = [
+        ('The parameter "deprecated_param_0" will be removed in Nilearn version 0.6.1rc. Please use the parameter "replacement_param_0" instead.'),
+        ('The parameter "deprecated_param_1" will be removed in Nilearn version 0.6.1rc. Please use the parameter "replacement_param_1" instead.'),
+        ]
+    with warnings.catch_warnings(record=True) as raised_warnings:
+        helpers._warn_deprecated_params(
+                replacement_params,
+                end_version='0.6.1rc',
```

Comment:
oops; we'll have to update this test for all releases ?

Reply:
No, it's merely checking that the end version is being displayed properly, the version I chose is arbitrary.